explorer/nodewatch/nodewatch/NetworkRepository.py

```diff
@@ -12,19 +12,52 @@
 class NodeDescriptor:
 	"""Node descriptor."""
 
-	def __init__(self, main_address=None, endpoint=None, name=None, version=None, height=0, finalized_height=0, balance=0, has_api=True):
+	# pylint: disable=too-many-instance-attributes
+
+	def __init__(
+		self,
+		main_address=None,
+		main_public_key=None,
+		endpoint=None,
+		name=None,
+		version=None,
+		height=0,
+		finalized_height=0,
+		balance=0,
+		roles=0xFF):
 		"""Creates a descriptor."""
 
 		# pylint: disable=too-many-arguments
 
 		self.main_address = main_address
+		self.main_public_key = main_public_key
 		self.endpoint = endpoint
 		self.name = name
 		self.version = version
 		self.height = height
 		self.finalized_height = finalized_height
 		self.balance = balance
-		self.has_api = has_api
+		self.roles = roles
```

Comment:
not sure if we need to add this now, but I guess adding 'node' key might be handy (for node key link txes)

Reply:
can add it

Reply:
notice this is not in normal peers information
and currently. due to way ingestion script, works it is not being retrieved for peer-only nodes

Reply:
hmm, you mean peer-only or having peer role? (the latter one would be fine ofc 😁 )

Reply:
peer only